### PR TITLE
Bugfix update calculate output function

### PR DESCRIPTION
**Pull request by Stevio54**
All CalculateOutput functions now accept parameters as needed.

### CODE REVIEW

File: src/Functions/Public/iaas/Get-vRACloudAccount.ps1 [hunk 1]
```diff
@@ -59,9 +59,9 @@
             $APIUrl = $APIUrl + "-$Type"
         }
 
-        function CalculateOutput {
+        function CalculateOutput([PSCustomObject]$CloudAccount, [String]$Type) {
 
-            if ($Type){
+            if ($Type -ne $nulls){
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRACloudAccount.ps1:64]: Should this be `$null`?

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 2]
```diff
@@ -171,9 +171,9 @@
                 'ById' {
                     if ($Force -or $PsCmdlet.ShouldProcess($Id)){
                         # --- Check to see if the DiskId's were optionally present
-                        $Response = Invoke-vRARestMethod -URI "$APIUrl`/$Id`/disks" -Method GET -Body $Body
+                        $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$Id`/disks" -Method GET -Body $Body
 
-                        CalculateOutput
+                        CalculateOutput $blockDeviceId $CompletionTimeout $WaitForCompletion $RestResponse
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:176]: Variable should be PascalCase

`$blockDeviceId` -> `$BlockDeviceId`

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 3]
```diff
@@ -185,9 +185,9 @@
                         $machineResponse = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$Name'`&`$select=id" -Method GET
                         $machineId = $machineResponse.content[0].id
 
-                        $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks" -Method POST -Body $Body
+                        $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks" -Method POST -Body $Body
 
-                        CalculateOutput
+                        CalculateOutput $blockDeviceId $CompletionTimeout $WaitForCompletion $RestResponse
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:190]: Variable should be PascalCase

`$blockDeviceId` ->` $BlockDeviceId`

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 4]
```diff
@@ -97,13 +97,13 @@
 
         $APIUrl = "/iaas/api/machines"
 
-        function CalculateOutput {
+        function CalculateOutput([int]$CompletionTimeout,[switch]$WaitForCompletion,[PSCustomObject]$RestResponse) {
             if ($WaitForCompletion) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:101]: Although not part of this PR, we should consider using IsPresent() when evaluating switch types as it will give a more reliable result.

File: src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1 [hunk 5]
```diff
@@ -243,7 +211,15 @@
...
                 $AssociatedCloudAccountIdsFormatForBodyText = $AssociatedCloudAccountIdsAddQuotes -join ","
 
-                $RegionIDs = ProcessReqionIds # process to see if regions were given, if not, we pull all of them
+                if ($null -eq $RegionIds) {
+                    # process to see if regions were given, if not, we pull all of them
+                    if ($PSBoundParameters.ContainsKey("Credential")){
+                        $RegionIDs = Get-vRARegionEnumerationvSphere -HostName $HostName  -Credential $Credential -AcceptSelfSignedCertificate:$($AcceptSelfSignedCertificate)
+                    } else {
+                        # assume username and password
+                        $RegionIDs = Get-vRARegionEnumerationvSphere -HostName $HostName  -username $username -Password $Password -AcceptSelfSignedCertificate:$($AcceptSelfSignedCertificate)
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1:220]: Variables should be PascalCase
`$username` -> `$Username`

File: src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1 [hunk 6]
```diff
@@ -75,7 +75,7 @@
 
                         $OnboardingDeployment= Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingDeployment)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1:78]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1 [hunk 7]
```diff
@@ -98,7 +98,7 @@
                             if ($OnboardingDeployment.name -eq $OnboardingDeploymentName){
 
                                 $MatchedOnboardingDeployment = $true
-                                CalculateOutput
+                                CalculateOutput($OnboardingDeployment)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1:101]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1 [hunk 8]
```diff
@@ -121,7 +121,7 @@
 
                         $OnboardingDeployment = $Response.documents.$document
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingDeployment)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingDeployment.ps1:124]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1 [hunk 9]
```diff
@@ -85,7 +85,7 @@
 
                         $OnboardingPlan= Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1:88]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1 [hunk 10]
```diff
@@ -108,7 +108,7 @@
                             if ($OnboardingPlan.name -eq $OnboardingPlanName){
 
                                 $MatchedOnboardingPlan = $true
-                                CalculateOutput
+                                CalculateOutput($OnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1:111]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1 [hunk 11]
```diff
@@ -131,7 +131,7 @@
 
                         $OnboardingPlan = $Response.documents.$document
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingPlan.ps1:134]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingPlanExecution.ps1 [hunk 12]
```diff
@@ -75,7 +75,7 @@
 
                         $ExecuteOnboardingPlan = Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($ExecuteOnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingPlanExecution.ps1:78]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingPlanExecution.ps1 [hunk 13]
```diff
@@ -87,7 +87,7 @@
 
                         $ExecuteOnboardingPlan = Invoke-vRARestMethod -Method GET -URI $OnboardingPlanExecutionLink -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($ExecuteOnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingPlanExecution.ps1:90]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1 [hunk 14]
```diff
@@ -78,7 +78,7 @@
                         $URI = "$($APIUrl)/$($OnboardingResourceId)"
                         $OnboardingResource= Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingResource)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1:81]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1 [hunk 15]
```diff
@@ -100,7 +100,7 @@
                             if ($OnboardingResource.resourceName -eq $OnboardingResourceName){
 
                                 $MatchedOnboardingResource = $true
-                                CalculateOutput
+                                CalculateOutput($OnboardingResource)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1:103]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1 [hunk 16]
```diff
@@ -122,7 +122,7 @@
 
                         $OnboardingResource = $Response.documents.$document
 
-                        CalculateOutput
+                        CalculateOutput($OnboardingResource)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Get-vRAOnboardingResource.ps1:125]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 [hunk 17]
```diff
@@ -107,7 +107,7 @@
 "@
                             $ExecuteOnboardingPlan= Invoke-vRARestMethod -Method POST -URI $URI -Body $Body -Verbose:$VerbosePreference
 
-                            CalculateOutput
+                            CalculateOutput($ExecuteOnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1:110]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 [hunk 18]
```diff
@@ -128,7 +128,7 @@
...
```
Comment (chelnak) [inline src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1:131]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 [hunk 19]
```diff
@@ -146,7 +146,7 @@
...
```
Comment (chelnak) [inline src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1:149]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 [hunk 20]
```diff
@@ -161,7 +161,7 @@
 
                         $ExecuteOnboardingPlan= Invoke-vRARestMethod -Method POST -URI $URI -Body $Body -Verbose:$VerbosePreference
 
-                        CalculateOutput
+                        CalculateOutput($ExecuteOnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1:164]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/New-vRAOnboardingDeployment.ps1 [hunk 21]
```diff
@@ -97,7 +97,7 @@
                 $URI = "/relocation/onboarding/deployment"
                 $OnboardingDeployment = Invoke-vRARestMethod -Method POST -URI $URI -Body $Body -Verbose:$VerbosePreference
 
-                CalculateOutput
+                CalculateOutput($OnboardingDeployment)
```
Comment (chelnak) [inline src/Functions/Public/relocation/New-vRAOnboardingDeployment.ps1:100]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/New-vRAOnboardingPlan.ps1 [hunk 22]
```diff
@@ -112,7 +112,7 @@
                 $URI = "/relocation/onboarding/plan"
                 $OnboardingPlan = Invoke-vRARestMethod -Method POST -URI $URI -Body $Body -Verbose:$VerbosePreference
 
-                CalculateOutput
+                CalculateOutput($OnboardingPlan)
```
Comment (chelnak) [inline src/Functions/Public/relocation/New-vRAOnboardingPlan.ps1:115]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/New-vRAOnboardingResource.ps1 [hunk 23]
```diff
@@ -141,7 +141,7 @@
                 $URI = "/relocation/onboarding/resource"
                 $OnboardingResource = Invoke-vRARestMethod -Method POST -URI $URI -Body $Body -Verbose:$VerbosePreference
 
-                CalculateOutput
+                CalculateOutput($OnboardingResource)
```
Comment (chelnak) [inline src/Functions/Public/relocation/New-vRAOnboardingResource.ps1:144]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/iaas/Get-vRACloudAccount.ps1 (hunk 1 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/Get-vRACloudAccount.ps1:64]: fixed in next commit.

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 (hunk 3 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:190]: fixed with a mass find/replace in next commit.

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 (hunk 4 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/Resize-vRAMachine.ps1:101]: I went through all of the code and made this change. Whenever a switch is evaluated, it is done so by using the IsPresent function

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 (hunk 19 above)
Comment (Stevio54) [reply src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1:149]: I went through and removed the parenthesis where I found them. Should be in next commit.

File: src/Functions/Public/relocation/New-vRAOnboardingDeployment.ps1 (hunk 21 above)
Comment (Stevio54) [reply src/Functions/Public/relocation/New-vRAOnboardingDeployment.ps1:100]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/relocation/New-vRAOnboardingResource.ps1 (hunk 23 above)
Comment (Stevio54) [reply src/Functions/Public/relocation/New-vRAOnboardingResource.ps1:144]: Should the parameter be positional here and not in ()?

File: src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1 (hunk 5 above)
Comment (chelnak) [reply src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1:220]: -username should be -Username

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 24]
```diff
@@ -169,25 +169,25 @@
...
                 # --- Get Machine by its name
                 # --- Will need to retrieve the machine first, then use ID to get final output
                 'ByName' {
-                    if ($Force -or $PsCmdlet.ShouldProcess($Name)){
+                    if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($Name)){
                         $machineResponse = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$Name'`&`$select=id" -Method GET
                         $machineId = $machineResponse.content[0].id
 
-                        $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks" -Method POST -Body $Body
+                        $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks" -Method POST -Body $Body
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:188]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAVirtualDisk.ps1 [hunk 25]
```diff
@@ -159,46 +159,46 @@ function New-vRAVirtualDisk {
...
                             $Body = @"
                                 {
                                     "capacityInGB": $($CapacityInGB),
-                                    "encrypted": $($Encrypted),
+                                    "encrypted": $($Encrypted.IsPresent),
                                     "name": "$($Name)",
                                     "description": "$($DeviceDescription)",
-                                    "persistent": $($Persistent),
+                                    "persistent": $($Persistent.IsPresent),
                                     "projectId": "$($projId)"
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAVirtualDisk.ps1:195]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 26]
```diff
@@ -163,9 +163,9 @@
                     'ResizeFlavorById' {
 
                         foreach ($machineId in $Id) {
-                            if ($Force -or $PsCmdlet.ShouldProcess($machineId)) {
-                                $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?name=$Flavor" -Method POST
-                                CalculateOutput
+                            if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($machineId)) {
+                                $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?name=$Flavor" -Method POST
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:167]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 27]
```diff
@@ -175,12 +175,12 @@
                     'ResizeFlavorByName' {
 
                         foreach ($machine in $Name) {
-                            if ($Force -or $PsCmdlet.ShouldProcess($machine)) {
+                            if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($machine)) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:178]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 28]
```diff
@@ -175,12 +175,12 @@
...
                             $machineResponse = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$machine'`&`$select=id" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:179]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 29]
```diff
@@ -175,12 +175,12 @@
                     'ResizeFlavorByName' {
 
                         foreach ($machine in $Name) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:177]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 30]
```diff
@@ -163,9 +163,9 @@
                     'ResizeFlavorById' {
 
                         foreach ($machineId in $Id) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:165]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 31]
```diff
@@ -175,12 +175,12 @@
                     'ResizeFlavorByName' {
 
                         foreach ($machine in $Name) {
-                            if ($Force -or $PsCmdlet.ShouldProcess($machine)) {
+                            if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($machine)) {
                             $machineResponse = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$machine'`&`$select=id" -Method GET
                             $machineId = $machineResponse.content[0].Id
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:180]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 32]
```diff
@@ -175,12 +175,12 @@
...
 
-                            $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?name=$Flavor" -Method POST
-                            CalculateOutput
+                            $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?name=$Flavor" -Method POST
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:182]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 33]
```diff
@@ -190,9 +190,9 @@
                     'ResizeById' {
 
                         foreach ($machineId in $Id) {
-                            if ($Force -or $PsCmdlet.ShouldProcess($machineId)) {
-                                $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?memoryInMB=$Memory`&cpuCount=$CPU" -Method POST
-                                CalculateOutput
+                            if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($machineId)) {
+                                $RestResponse = Invoke-vRARestMethod -URI "$APIUrl`/$machineId/operations/resize?memoryInMB=$Memory`&cpuCount=$CPU" -Method POST
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:194]: Variables should be PascalCase

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 [hunk 34]
```diff
@@ -202,12 +202,12 @@
                     'ResizeByName' {
 
                         foreach ($machine in $Name) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Resize-vRAMachine.ps1:204]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1 [hunk 35]
```diff
@@ -76,7 +76,7 @@
                 'ById' {
                     foreach ($executionId in $Id) {
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=id eq '$executionId'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1:78]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1 [hunk 36]
```diff
@@ -86,7 +86,7 @@
                 'ByName' {
                     foreach ($pipelineName in $Pipeline) {
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$pipelineName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1:88]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1 [hunk 37]
```diff
@@ -96,7 +96,7 @@
                 'ByProject' {
                     foreach ($projectName in $Project) {
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1:97]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1 [hunk 38]
```diff
@@ -96,7 +96,7 @@
...
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=project eq '$projectName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamExecution.ps1:98]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1 [hunk 39]
```diff
@@ -75,7 +75,7 @@
                 'ById' {
                     foreach ($pipelineId in $Id) {
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=id eq '$pipelineId'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1:77]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1 [hunk 40]
```diff
@@ -85,7 +85,7 @@
                 'ByName' {
                     foreach ($pipelineName in $Pipeline) {
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1:86]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1 [hunk 41]
```diff
@@ -85,7 +85,7 @@
...
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$pipelineName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1:87]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1 [hunk 42]
```diff
@@ -95,7 +95,7 @@
                 'ByProject' {
                     foreach ($projectName in $Project) {
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=project eq '$projectName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamPipeline.ps1:97]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1 [hunk 43]
```diff
@@ -76,7 +76,7 @@
                 'ById' {
                     foreach ($variableId in $Id) {
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=id eq '$variableId'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1:78]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1 [hunk 44]
```diff
@@ -86,7 +86,7 @@
                 'ByName' {
                     foreach ($variableName in $Variable) {
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1:87]: Variables should be PascalCase

File: src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1 [hunk 45]
```diff
@@ -86,7 +86,7 @@
...
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$variableName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/codestream/Get-vRACodeStreamVariable.ps1:88]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachine.ps1 [hunk 46]
```diff
@@ -72,7 +72,7 @@
                 'ById' {
                     foreach ($machineId in $Id) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachine.ps1:73]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachine.ps1 [hunk 47]
```diff
@@ -72,7 +72,7 @@
...
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=id eq '$machineId'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachine.ps1:74]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachine.ps1 [hunk 48]
```diff
@@ -82,7 +82,7 @@
                 'ByName' {
                     foreach ($machineName in $Name) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachine.ps1:83]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachine.ps1 [hunk 49]
```diff
@@ -82,7 +82,7 @@
...
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$machineName'" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachine.ps1:84]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachineDisk.ps1 [hunk 50]
```diff
@@ -143,7 +143,7 @@
 
                             $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks`/$disk" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachineDisk.ps1:144]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRAMachineDisk.ps1 [hunk 51]
```diff
@@ -152,7 +152,7 @@
 
                         $Response = Invoke-vRARestMethod -URI "$APIUrl`/$machineId`/disks" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRAMachineDisk.ps1:153]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRARegionEnumerationvSphere.ps1 [hunk 52]
```diff
@@ -0,0 +1,122 @@
...
+    Param (
+
+        [Parameter(Mandatory = $false, ParameterSetName = "Username")]
+        [Parameter(Mandatory = $false, ParameterSetName = "Credential")]
+        [ValidateNotNullOrEmpty()]
+        [String]$HostName,
+
+        [Parameter(Mandatory = $true, ParameterSetName = "Username")]
+        [ValidateNotNullOrEmpty()]
+        [String]$username,
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRARegionEnumerationvSphere.ps1:50]: Variables should be PascalCase

File: src/Functions/Public/iaas/Get-vRARegionEnumerationvSphere.ps1 [hunk 53]
```diff
@@ -0,0 +1,122 @@
...
+    process {
+        # --- Create new Azure Cloud Account
+        try {
+
+            $Body = @"
+            {
+                "hostName": "$($HostName)",
+                "acceptSelfSignedCertificate": $($AcceptSelfSignedCertificateStatus),
+                "password": "$($JSONPassword)",
+                "username": "$($username)"
```
Comment (chelnak) [inline src/Functions/Public/iaas/Get-vRARegionEnumerationvSphere.ps1:102]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 54]
```diff
@@ -90,14 +90,14 @@
 
         $APIUrl = "/iaas/api/machines"
 
-        function CalculateOutput {
+        function CalculateOutput([String[]]$blockDeviceId,[int]$CompletionTimeout,[switch]$WaitForCompletion,[PSCustomObject]$RestResponse) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:93]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 55]
```diff
@@ -90,14 +90,14 @@
...
-            if ($WaitForCompletion) {
+            if ($WaitForCompletion.IsPresent) {
                 # if the wait for completion flag is given, the output will be different, we will wait here
                 # we will use the built-in function to check status
                 $elapsedTime = 0
                     do {
-                        $RequestResponse = Get-vRARequest -RequestId $Response.id
+                        $RequestResponse = Get-vRARequest -RequestId $RestResponse.id
                         if ($RequestResponse.Status -eq "FINISHED") {
                             foreach ($resource in $RequestResponse.Resources) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:102]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 56]
```diff
@@ -90,14 +90,14 @@
...
                                 $Response = Invoke-vRARestMethod -URI "$resource/disks/$blockDeviceId" -Method GET
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:103]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 57]
```diff
@@ -132,22 +132,22 @@
                     if ($elapsedTime -gt $CompletionTimeout -or $elapsedTime -eq $CompletionTimeout) {
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:132]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 [hunk 58]
```diff
@@ -169,25 +169,25 @@
...
                     break
                 }
 
                 # --- Get Machine by its name
                 # --- Will need to retrieve the machine first, then use ID to get final output
                 'ByName' {
-                    if ($Force -or $PsCmdlet.ShouldProcess($Name)){
+                    if ($Force.IsPresent -or $PsCmdlet.ShouldProcess($Name)){
                         $machineResponse = Invoke-vRARestMethod -URI "$APIUrl`?`$filter=name eq '$Name'`&`$select=id" -Method GET
                         $machineId = $machineResponse.content[0].id
```
Comment (chelnak) [inline src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:186]: Variables should be PascalCase

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 (hunk 2 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1]: If you look at the latest commit, this is already fixed.

File: src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1 (hunk 5 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1:220]: Will be fixed next commit :)

File: src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1 (hunk 20 above)
Comment (Stevio54) [reply src/Functions/Public/relocation/Invoke-vRAOnboardingPlan.ps1]: resolved in last commit.

File: src/Functions/Public/relocation/New-vRAOnboardingPlan.ps1 (hunk 22 above)
Comment (Stevio54) [reply src/Functions/Public/relocation/New-vRAOnboardingPlan.ps1]: resolved in last commit. :)

File: src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1 (hunk 24 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/New-vRAMachineAttachedDisk.ps1:188]: I think I got everything PascalCased

File: src/Functions/Public/iaas/Resize-vRAMachine.ps1 (hunk 27 above)
Comment (Stevio54) [reply src/Functions/Public/iaas/Resize-vRAMachine.ps1:178]: I think I got everything PascalCased

**Comment by jonathanmedd** [comment]
Resolves #225
